### PR TITLE
Fix flaky effectBlockingInterrupt

### PR DESCRIPTION
**Pull request by jdegoes**

### CODE REVIEW

**Comment by adamgfraser** [comment]
@jdegoes I'm still observing some flakiness on this test when I run it with `nonFlaky`.

**Comment by jdegoes** [comment]
@adamgfraser On it.

**Comment by jdegoes** [comment]
There were 2 sources of flakiness. Hopefully this does it.

**Comment by adamgfraser** [comment]
Still seeing some flakiness.

**Comment by jdegoes** [comment]
Reproduced locally with n = 1000 and greater.

**Comment by jdegoes** [comment]
I think I see it...

**Comment by adamgfraser** [comment]
Passing multiple iterations with n = 10000 now.

File: core/jvm/src/main/scala/zio/blocking/package.scala [hunk 1]
```diff
@@ -105,35 +108,38 @@ package object blocking {
...
+                              case _: InterruptedException =>
+                                Thread.interrupted // Clear interrupt status
+                                ZIO.interrupt
+                              case t: Throwable =>
+                                ZIO.failNow(t)
+                            } finally {
+                              withMutex { thread.set(None); end.set(()) }
+                            }
+                          }.forkDaemon
+                  a <- restore(fiber.join).ensuring(interruptThread)
```
Comment (adamgfraser) [inline core/jvm/src/main/scala/zio/blocking/package.scala]:
```suggestion
                  a <- restore(fiber.join.refailWithTrace).ensuring(interruptThread)
```

**Pull request merged**